File: DSA/vaishnavi_joshi_590016947/Day97/Q2.java
import java.util.*;

class Solution {
    public static int maxEnergy(int[] energy) {

        int n = energy.length;

        // dp[i] = maximum energy collected up to floor i
        int[] dp = new int[n];

        // Starting from floor 1
        dp[0] = energy[0];

        // Floor 2
        dp[1] = energy[1] + dp[0];

        // Calculate for remaining floors
        for (int i = 2; i < n; i++) {
            dp[i] = energy[i] + Math.max(dp[i - 1], dp[i - 2]);
        }

        return dp[n - 1];
    }

    public static void main(String[] args) {

        int[] energy = {5, 10, 20, 15, 25, 30};

        System.out.println(maxEnergy(energy));
    }
}

File: DSA/VivaswanSingh_590023022/Day_97/q2.c
#include <stdio.h>

int maxEnergy(int N, int energy[])
{
    if (N == 1)
        return energy[0];

    int prev2 = energy[0];
    int prev1 = energy[1] + energy[0];

    for (int i = 2; i < N; i++)
    {
        int current = energy[i] + (prev1 > prev2 ? prev1 : prev2);

        prev2 = prev1;
        prev1 = current;
    }

    return prev1;
}

int main()
{
    int N = 6;
    int energy[] = {5, 10, 20, 15, 25, 30};

    printf("Maximum energy: %d\n", maxEnergy(N, energy));

    return 0;
}